tsl/src/fdw/modify_plan.c

```diff
@@ -144,6 +108,62 @@ fdw_plan_foreign_modify(PlannerInfo *root, ModifyTable *plan, Index result_relat
 	if (plan->returningLists)
 		returning_list = (List *) list_nth(plan->returningLists, subplan_index);
 
+	/*
+	 * Core code already has some lock on each rel being planned, so we can
+	 * use NoLock here.
+	 */
+	rel = table_open(rte->relid, NoLock);
+	TupleDesc tupdesc = RelationGetDescr(rel);
+
+	/*
+	 * In an INSERT, we transmit all columns that are defined in the foreign
+	 * table.  In an UPDATE, if there are BEFORE ROW UPDATE triggers on the
+	 * foreign table, we transmit all columns like INSERT; else we transmit
+	 * only columns that were explicitly targets of the UPDATE, so as to avoid
+	 * unnecessary data transmission.  (We can't do that for INSERT since we
+	 * would miss sending default values for columns not listed in the source
+	 * statement, and for UPDATE if there are BEFORE ROW UPDATE triggers since
+	 * those triggers might change values for non-target columns, in which
+	 * case we would miss sending changed values for those columns.)
+	 */
+	if (operation == CMD_INSERT ||
```

Comment:
Any particular reason why the updated code was inlined as opposed to just updating the functions `get_insert_attrs` and `get_update_attrs` above?

Reply:
The `tsl/src/fdw/modify_plan.c:fdw_plan_foreign_modify` is copied from `postgres_fdw/postgres_fdw.c:postgresPlanForeignModify` so I keep it inline just to make comparison easy and also because the functions are not used anywhere so IMHO we don't need. And we'll eventually remove MN support so my main goal here is keep this code working.